contracts/ArenaToken.sol

```diff
@@ -96,6 +96,10 @@ contract ArenaToken is ERC20, ERC20Burnable, Ownable, ERC20Permit, ERC20Votes {
 
         // transfer claimable proportion to caller
         _transfer(address(this), msg.sender, claimableAmount);
+        // self-delegate if no prior delegatee was chosen
+        if (delegates(msg.sender) == address(0)) {
+            _delegate(msg.sender, msg.sender);
+        }
 
         require(address(tokenLock) != address(0), "Vesting contract not initialized");
         tokenLock.setupVesting(
```

Comment:
It might be useful to be able to use your vested tokens to vote.
(for example the graph allows you to delegate your vested tokens).
This would require some extra logic.

Perhaps it is useful to be able to delegate your tokens while claiming them (like ENS allows), this would require an extra parameter in claimTokens (like ENS has).
This can also be combined with vesting, but if you want to change the delegation while being vesting also requires extra logic.

By the way, "unchecked" isn't that important on polygon as the gas costs are a lot lower

Reply:
Also might be useful to add the delegation logic to 
https://github.com/code-423n4/contracts/blob/main/contracts/ArenaTokenSale.sol

Reply:
I think this was discussed internally, and we ended up sticking to to the current implementation.
The problem is that we're using 1 contract for everyone's vesting schedule... Since u can't delegate to multiple parties, the contracts will have to be modified to deploy a token lock per user.

+1 on adding delegation logic to the token sale contract.

Reply:
Yes, we initially even implemented being able to vote with the vested token but removed this logic after some discussion.
I don't think adding delegation logic to the token sale contract is that straightforward as only the msg.sender can change their delegation. we'd need to either add delegation-by-signature to the `buy` action (we don't have the resources to build a UI for this and were asked to already remove the `amount` parameter from this action, so this is likely not going to happen) or change the `ArenaToken` to give privileged access to change delegations for the token sale. I wouldn't want to change that either atm, the few whitelisted buyers can be told to use tally's UI to delegate after purchasing the tokens

Reply:
Ahhh right, I forgot `_delegate()` is an internal function XD. Current changes LGTM then.